### PR TITLE
fix(ClearRefinements): use connector transformItems

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

This migrates ClearRefinements to use the transformItems provided by the connector.
- improves the typing
- changes `createURL` default to `() => '#'`
- adds a storybook example for transformItems

⚠️ if you think there are too many changes, I can split it up.

ref https://github.com/algolia/angular-instantsearch/issues/372

### CODE REVIEW

**Comment by francoischalifour** [comment]
[Link to story →](https://deploy-preview-561--angular-instantsearch.netlify.com/examples/storybook/?path=/story/currentrefinements--with-includedattributes-query-brand)

File: src/clear-refinements/clear-refinements.ts [hunk 1]
```diff
@@ -28,15 +34,15 @@ export class NgAisClearRefinements extends BaseWidget {
   // instance options
   @Input() public includedAttributes: string[];
   @Input() public excludedAttributes: string[];
-  // TODO: add transformItems
+  @Input() public transformItems?: <U extends string>(items: string[]) => U[];
```
Comment (francoischalifour) [inline src/clear-refinements/clear-refinements.ts]: What's the benefit of extending `string` here?

Comment (tkrugg) [reply src/clear-refinements/clear-refinements.ts]: You're there is not benefit. I'll remove it.

Comment (tkrugg) [reply src/clear-refinements/clear-refinements.ts]: fixed here: a4d7f36

**Pull request merged**